bm-persona/Map/MapViewController.swift

```diff
@@ -167,19 +198,44 @@ extension MapViewController: MKMapViewDelegate {
             annotationView.annotation = marker
             annotationView.image = marker.type.icon()
             return annotationView
+        } else if let searchAnnotation = annotation as? SearchAnnotation,
+            // create new pin on map for searched item
+            let annotationView = mapView.dequeueReusableAnnotationView(withIdentifier: MKMapViewDefaultAnnotationViewReuseIdentifier) as? MKMarkerAnnotationView {
+            annotationView.annotation = searchAnnotation
+            annotationView.glyphImage = searchAnnotation.icon()
+            annotationView.contentMode = .scaleToFill
+            annotationView.markerTintColor = searchAnnotation.color()
+            annotationView.glyphTintColor = .white
+            return annotationView
         }
         return MKAnnotationView()
     }
     
     func mapView(_ mapView: MKMapView, didSelect view: MKAnnotationView) {
-        markerDetail.marker = view.annotation as? MapMarker
-        drawerContainer?.moveDrawer(to: .hidden, duration: 0.2)
+        if let annotation = view.annotation as? MapMarker {
+            markerDetail.marker = annotation
+            // hide either search detail or drawer
+            if detailViewController != nil {
+                detailViewController!.view.isHidden = true
```

Comment:
Small nitpick, but when hiding / showing `detailViewController`, can we animate it down / up like the main drawer?

Reply:
yeah I tried to do that before and tried again, but there's a problem and idk how to fix it. basically, calling moveDrawer(.hidden) moves it to hidden but something automatically moves it back to the middle position. asynchronously calling moveDrawer(.hidden) works but when selecting a second mapMarker, you can see the view flash up then down, so it looks like the view is moving up for some reason. I set a breakpoint in moveDrawer, but it only gets called once so I've got no clue why the view moves twice

Reply:
so there were some other problems with the movedrawer function, but essentially I tried moving the center of the drawer twice, once initially and once in the completion handler. somehow, this fixed this bug and some others. not sure why